pkg/gatherers/conditional/validation.go

```diff
@@ -0,0 +1,111 @@
+package conditional
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/go-playground/validator"
+)
+
+var validate *validator.Validate
+
+func init() {
+	validate = validator.New()
+
+	validate.RegisterStructValidation(gatheringRuleValidator, GatheringRule{})
+	validate.RegisterAlias("openshift_namespace", "min=1,max=128,hostname,startswith=openshift-")
+}
+
+func validateGatheringRules(gatheringRules []GatheringRule) []error {
+	var allErrors []error
+
+	for _, gatheringRule := range gatheringRules {
+		errs := validateGatheringFunctionsParamsAreStructs(gatheringRule.GatheringFunctions)
+		if len(errs) > 0 {
+			allErrors = append(allErrors, errs...)
+			continue
+		}
+
+		err := validate.Struct(gatheringRule)
+		if err != nil {
+			allErrors = append(allErrors, err)
+			continue
+		}
+	}
+
+	return allErrors
+}
+
+func validateGatheringFunctionsParamsAreStructs(gatheringFunctions GatheringFunctions) (errs []error) {
+	// a workaround to not panic https://github.com/go-playground/validator/issues/789
+	for key, val := range gatheringFunctions {
+		if reflect.TypeOf(val).Kind() != reflect.Struct {
```

Comment:
I think this type of validation should be easily enforced with validating the JSON condition definition against JSON schema. My feeling is that we started from the wrong end here. I think we will receive JSON data from the new service so I guess these kinds of problems will be caught when doing `json.Unmarshal`.

Reply:
Well, this PR lacks the unmarshalling thing, but it will be later added and it'll be not too much of code, I just wanted to introduce it in one of the next PRs. Yes, it's doing some extra validation, but it may be useful if we for example decide to keep some rules in the code of the operator and some in the external repo.

Reply:
OK, but this option (having something in the code and something externally) now seems like a bit of overengineering to me TBH. I would leave this kind of validation up to the JSON schema (we can validate here against it as well).

Reply:
Hm, not really sure if the schema validation can do everything we need, I will take the task about schema validation for the repo and explore if it works for the operator too.

Reply:
Me neither. I am just saying I would do as much as possible in the schema and leave some leftover here in the code. Because imagine one wants to introduce a new condition type. In this situation you would need an update in the IO code (possibly bugzilla, backporting, all the annoying stufff etc.) to pass the validation vs just updating the schema (which can be retrieved from the repository). 
It's just an idea and needs some real proof/verification.